src/Streamly/Internal/FileSystem/Handle.hs

```diff
@@ -257,9 +257,32 @@ readChunksWithBufferOf = Unfold step return
 --
 --
 {-# INLINE_NORMAL readChunksFromToWith #-}
-readChunksFromToWith :: -- MonadIO m =>
+readChunksFromToWith :: MonadIO m =>
     Unfold m (Int, Int, Int, Handle) (Array Word8)
-readChunksFromToWith = undefined
+readChunksFromToWith = Unfold step inject
+
+    where
+
+    inject (from, to, buffSize, h) = do
+        liftIO $ hSeek h AbsoluteSeek $ fromIntegral from
+        return (to - from + 1, buffSize, h)
+
+    {-# INLINE_LATE step #-}
+    step (remaining, buffSize, h) =
+        if remaining <= 0
+        then return D.Stop
+        else
+            do
+            arr <- getChunk (min buffSize remaining) h
+            return $
+                case A.length arr of
+                    0 -> D.Stop
+                    len ->
+                        if (remaining - len) >= 0
+                        then
+                            D.Yield arr (remaining - len, buffSize, h)
+                        else
+                            error "Bug: ..."
```

Comment:
I meant some message in place of `...`, e.g. ` error "Bug: readFromChunksFromToWith: array length more than expected`.